src/parsed/element_type.rs

```diff
@@ -26,21 +28,23 @@ impl ElementType {
         let code = blob.read_unsigned();
         match code {
```

Comment:
You may want to consider using `std::convert::TryFrom<&mut Blob>`  instead of having a custom function that panics if it doesn't recognize the code.